cocotb/share/include/cocotb_utils.h

```diff
@@ -42,8 +44,23 @@ extern void* utils_dyn_sym(void *handle, const char* sym_name);
 
 extern int is_python_context;
 
-void to_python(void);
-void to_simulator(void);
+#define to_python() do { \
+    if (is_python_context) { \
+        LOG_ERROR("FATAL: We are calling up again"); \
+        exit(1); \
+    } \
+    ++is_python_context; \
+    LOG_DEBUG("Returning to Python"); \
+} while (0)
+
+#define to_simulator() do { \
+    if (!is_python_context) { \
+        LOG_ERROR("FATAL: We have returned twice from python\n"); \
+        exit(1); \
+    } \
+    --is_python_context; \
+    LOG_DEBUG("Returning to simulator"); \
+} while (0)
```

Comment:
I assume these are macros so that the filename and line number is more useful in the log message? If so, can you add a comment explaining that?